File: app/src/main/java/com/lucasurbas/counter/app/di/FragmentModule.java
package com.lucasurbas.counter.app.di;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;

import dagger.Module;
import dagger.Provides;
import com.lucasurbas.counter.app.di.scope.FragmentScope;

@Module
public class FragmentModule {

    private final Fragment fragment;

    public FragmentModule(@NonNull final Fragment fragment) {
        this.fragment = fragment;
    }

    @Provides
    @FragmentScope
    Fragment provideFragment() {
        return fragment;
    }
}

File: app/src/main/java/com/lucasurbas/counter/app/di/FragmentModule.kt
package com.lucasurbas.counter.app.di

import android.support.v4.app.Fragment
import com.lucasurbas.counter.app.di.scope.FragmentScope
import dagger.Module
import dagger.Provides

@Module
class FragmentModule(private val fragment: Fragment) {

    @Provides
    @FragmentScope
    internal fun provideFragment(): Fragment {
        return fragment
    }
}
